Resources/public/js/views/ez-dashboardblockbasedview.js

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-dashboardblockbasedview', function (Y) {
+    'use strict';
+    /**
+     * Provides the Dashboard Block Based View class
+     *
+     * @module ez-dashboardblockbasedview
+     */
+    Y.namespace('eZ');
+
+    var SELECTOR_CONTENT = '.dashboard-content';
+
+    /**
+     * The dashboard view
+     *
+     * @namespace eZ
+     * @class DashboardBlockBasedView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.DashboardBlockBasedView = Y.Base.create('dashboardBlockBasedView', Y.eZ.TemplateBasedView, [], {
+        initializer: function () {
+            this.onceAfter('activeChange', function (event) {
+                this.after('blocksChange', this._renderBlocks, this);
+            }, this);
+
+            this.after('activeChange', this._toggleBlocksActiveState, this);
+        },
+
+        /**
+         * Renders the dashboard view
+         *
+         * @method render
+         * @return {eZ.DashboardBlockBasedView} the view itself
+         */
+        render: function () {
+            this.get('container').setHTML(this.template());
+            this._renderBlocks();
+
+            return this;
+        },
+
+        /**
+         * Toggles blocks active state
+         *
+         * @method _toggleBlocksActiveState
+         * @protected
+         * @param event {Object} event facade
+         */
+        _toggleBlocksActiveState: function (event) {
+            this.get('blocks').forEach(function (block) {
+                block.set('active', event.newVal);
+            });
+        },
+
+        /**
+         * Renders blocks on the dashboard
+         *
+         * @method _renderBlocks
+         * @protected
+         */
+        _renderBlocks: function () {
+            var fragment = Y.one(document.createDocumentFragment());
+
+            this.get('blocks').forEach(function (block) {
+                fragment.append(block.render().get('container'));
+            });
+
+            this.get('container').one(SELECTOR_CONTENT).setHTML(fragment);
+        },
+
+        /**
+         * Sorts the provided list of blocks by priority.
+         *
+         * @method _sortBlocks
+         * @protected
+         * @param blocks {Array} list of blocks
+         * @return {Array} sorted list of blocks
+         */
+        _sortBlocks: function (blocks) {
+            return blocks.sort(function (a, b) {
+                return a.get('priority') > b.get('priority');
```

Comment:
according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort the compare function should return a number not a boolean so you should return `b.get('priority') - a.get('priority')` instead (higher priority first).